js/modules/k6/experimental/csv/module.go

```diff
@@ -0,0 +1,289 @@
+		var err error
+		options, err = newParserOptionsFrom(call.Argument(1).ToObject(rt))
+		if err != nil {
+			common.Throw(rt, fmt.Errorf("encountered an error while interpreting Parser options; reason: %w", err))
+		}
+	}
+
+	// Instantiate and configure a csv reader using the provided file and options
+	r, err := NewReaderFrom(file.ReadSeekStater, options)
+	if err != nil {
+		common.Throw(rt, fmt.Errorf("failed to create a new parser; reason: %w", err))
+	}
+
+	// Create a new Parser instance
+	parser := Parser{
+		reader:  r,
+		options: options,
+		vu:      mi.vu,
+	}
+
+	return rt.ToValue(&parser).ToObject(rt)
+}
+
+// Next returns the next row in the CSV file.
+func (p *Parser) Next() *sobek.Promise {
+	promise, resolve, reject := promises.New(p.vu)
+
+	go func() {
+		var records []string
+		var done bool
+		var err error
+
+		records, err = p.reader.Read()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				done = true
+			} else {
+				reject(err)
+				return
+			}
+		}
+
+		p.CurrentLine.Add(1)
+
+		resolve(parseResult{Done: done, Value: records})
+	}()
+
+	return promise
+}
+
+// parseResult holds the result of a CSV parser's parsing operation such
+// as when calling the [Next].
+type parseResult struct {
+	// Done indicates whether the parser has finished reading the file.
+	Done bool `js:"done"`
+
+	// Value holds the line's records value.
+	Value []string `js:"value"`
+}
+
+// options holds options used to configure CSV parsing when utilizing the module.
+//
+// The options can be used to either configure the CSV parser, or the parse function.
+// They offer to customize the behavior of the parser, such as the delimiter, whether
+// to skip the first line, or  to start reading from a specific line, and stop reading
+// at a specific line.
+type options struct {
+	// Delimiter is the character that separates the fields in the CSV.
+	Delimiter rune `js:"delimiter"`
+
+	// SkipFirstLine indicates whether the first line should be skipped.
+	SkipFirstLine bool `js:"skipFirstLine"`
+
+	// FromLine indicates the line from which to start reading the CSV file.
+	FromLine null.Int `js:"fromLine"`
+
+	// ToLine indicates the line at which to stop reading the CSV file (inclusive).
+	ToLine null.Int `js:"toLine"`
+}
+
+// newDefaultParserOptions creates a new options instance with default values.
+func newDefaultParserOptions() options {
+	return options{
+		Delimiter:     ',',
+		SkipFirstLine: false,
+	}
+}
+
+// newParserOptions creates a new options instance from the given Goja object.
+func newParserOptionsFrom(obj *sobek.Object) (options, error) {
+	// Initialize default options
+	options := options{
+		Delimiter:     ',',
+		SkipFirstLine: false,
+	}
+
+	if obj == nil {
+		return options, nil
+	}
+
+	if v := obj.Get("delimiter"); v != nil {
+		delimiter := v.String()
+
+		// A delimiter is gonna be treated as a rune in the Go code, so we need to make sure it's a single character.
+		if len(delimiter) > 1 {
```

Comment:
Fair point! Perhaps we should add a test with that scenario, and evaluate how could be handled?

Unless it's super tricky, because in such case I'd suggest to ship an initial release explicitly stating that limitation. But I hope that's not the case.

Reply:
To be sure I don't misunderstand, a matching delimiter would be something along the lines of `\t,` or `\t,\t`?

Reply:
@oleiade I mean, the delimiter can have a length of more than one rune like in the case of tab-separated CSV, it's `\t`

Reply:
Then I'm not sure this is an actual problem, as `\t` is a single character and fits in a single run (ascii decimal code 9) 👀 Or am I missing something?

```go
fmt.Println(rune('\t'))  // is valid
```

Reply:
@oleiade yes, you're right, resolving the tread :+1: